test/cross-chain/cross-chain.js

```diff
@@ -148,4 +457,992 @@ contract("Cross-chain", () => {
+      await client.confirmNewHash();
+
+      // Check in state
+      const key = await ReputationMinerTestWrapper.getKey(foreignColony.address, entry.skillId, accounts[0]);
+      expect(client.reputations[key]).to.not.equal(undefined);
+      expect(ethers.BigNumber.from(client.reputations[key].slice(0, 66)).toHexString()).to.equal("0x1337");
+
+      // Bridge it
+
+      p = bridgeMonitor.getPromiseForNextBridgedTransaction();
+      const tx = await homeColonyNetwork.bridgeCurrentRootHash(foreignChainId);
+      await tx.wait();
+      await p;
+
+      // Check state bridged to host chain
+      const foreignChainRootHash = await foreignColonyNetwork.getReputationRootHash();
+      const foreignNLeaves = await foreignColonyNetwork.getReputationRootHashNNodes();
+      const homeChainRootHash = await homeColonyNetwork.getReputationRootHash();
+      const homeNLeaves = await homeColonyNetwork.getReputationRootHashNNodes();
+
+      expect(foreignChainRootHash).to.equal(homeChainRootHash);
+      expect(homeNLeaves.toHexString()).to.equal(foreignNLeaves.toHexString());
+    });
+
+    it("if bridge disabled, reputation emissions are stored to be reemitted later", async () => {
+      let tx = await foreignBridge.setBridgeEnabled(false);
+      await tx.wait();
+      const bridgedReputationUpdateCountBefore = await foreignColonyNetwork.getBridgedReputationUpdateCount(foreignChainId, foreignColony.address);
+      tx = await foreignColony.emitDomainReputationReward(1, accounts[0], "0x1337");
+      await tx.wait();
+
+      // See it was stored for later
+      const bridgedReputationUpdateCountAfter = await foreignColonyNetwork.getBridgedReputationUpdateCount(foreignChainId, foreignColony.address);
+      expect(bridgedReputationUpdateCountAfter.sub(bridgedReputationUpdateCountBefore).toNumber()).to.equal(1);
+    });
+
+    it("if bridge disabled, cannot bridge current state", async () => {
+      let tx = await homeBridge.setBridgeEnabled(false);
+      await tx.wait();
+      tx = await homeColonyNetwork.bridgeCurrentRootHash(foreignChainId, { gasLimit: 1000000 });
+      await checkErrorRevertEthers(tx.wait(), "colony-mining-bridge-call-failed");
+    });
+
+    it("if bridge not set, cannot bridge current state", async () => {
+      let tx = await homeMetacolony.setColonyBridgeAddress(ADDRESS_ZERO);
+      await tx.wait();
+      tx = await homeColonyNetwork.bridgeCurrentRootHash(foreignChainId, { gasLimit: 1000000 });
+      await checkErrorRevertEthers(tx.wait(), "colony-network-bridge-not-set");
+    });
+
+    it("if bridge unknown, cannot bridge current state", async () => {
+      const tx = await homeColonyNetwork.bridgeCurrentRootHash(ADDRESS_ZERO, { gasLimit: 1000000 });
+      await checkErrorRevertEthers(tx.wait(), "colony-bridge-not-known-chain");
+    });
+
+    it("stored reputation emissions can be emitted later", async () => {
+      let tx = await foreignBridge.setBridgeEnabled(false);
```

Comment:
Why are we `await`ing these transactions twice?

Reply:
These are ethers contracts, so the first promise [resolves to a TransactionResponse](https://docs.ethers.org/v5/api/contract/contract/#contract-functionsSend) object. This only indicates it has been sent to the network. The `wait` property on the transaction response waits for the receipt, which indicates it has been mined.